### DISCUSSION TITLE
Remove identifier in ledger keys.

### DISCUSSION THREAD

**Discussion by AurelienFT** [massalabs/massa#3056] [Ideas]
Currently, in ledger keys we save an identifier to distinguish if the key is a balance, a bytecode or a datastore entry.

With  https://github.com/massalabs/massa/discussions/2970 the bytecode will dissapear. We could now remove the identifier from key and say that if the key size == address size then it's a balance otherwise if the key is longer it's a datastore key.

This will also help in storage costs to avoid people paying for the storage of this byte.

**Comment by damip** [comment]
That would be nice !
Note however that the bytecode-in-datastore rfc was not accepted yet (only the datastore in executesc part was)
